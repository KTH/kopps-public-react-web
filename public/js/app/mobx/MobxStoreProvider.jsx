/* eslint no-use-before-define: ["error", "nofunc"] */

import React from 'react'

import { createMobxManager } from './manager'

export default MobxStoreProvider

/**
 * React-component which prepares a MobX-store
 * so that its data can be used with the useStore() hook
 * of any other component embedded into this Provider.
 *
 * @param {object} props
 * @param {function} props.initCallback
 * @param {string} [props.storeId]
 * @param {object} props.children
 *
 */
function MobxStoreProvider(props) {
  const { initCallback, storeId, children } = props

  const manager = createMobxManager(storeId || 'default', initCallback)

  const { Provider: ReactContextProvider, value: mobxStore } = manager.getProviderComponentAndValue()

  return <ReactContextProvider value={mobxStore}>{children}</ReactContextProvider>
}
