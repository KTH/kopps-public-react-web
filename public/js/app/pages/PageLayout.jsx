import React from 'react'
import { Col, Row } from 'reactstrap'
import MainMenu from '../components/MainMenu'

function MainContent({ children }) {
  return <main id="mainContent">{children}</main>
}

function PageLayout({ children }) {
  return (
    <Row style={{ padding: '30px 0' }}>
      <MainMenu />
      <Col>
        <MainContent>{children}</MainContent>
      </Col>
    </Row>
  )
}

export default PageLayout
