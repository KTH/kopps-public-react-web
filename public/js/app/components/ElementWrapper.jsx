/* eslint-disable react/jsx-props-no-spreading */
import React from 'react'
import PropTypes from 'prop-types'
import { MobxStoreProvider } from '../mobx'
import NotFound from '../pages/NotFound'

function ElementWrapper({ component: Component, layout: Layout, applicationStore, createMenuData }) {
  const menuData = createMenuData(applicationStore)
  const { statusCode } = applicationStore

  React.useEffect(() => {
    let isMounted = true
    if (isMounted) {
      const siteNameElement = document.querySelector('.block.siteName a')
      if (siteNameElement) {
        const rawHref = siteNameElement.href
        siteNameElement.href = rawHref.replace('.se//', '.se/')
      }
    }
    return () => (isMounted = false)
  }, [])

  return (
    <MobxStoreProvider initCallback={() => applicationStore}>
      <Layout menuData={menuData}>{statusCode === 404 ? <NotFound /> : <Component />}</Layout>
    </MobxStoreProvider>
  )
}

ElementWrapper.propTypes = {
  component: PropTypes.oneOfType([PropTypes.node, PropTypes.func, PropTypes.elementType]).isRequired,
  layout: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
  createMenuData: PropTypes.func.isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  applicationStore: PropTypes.oneOfType([PropTypes.object, PropTypes.func]).isRequired,
}

export default ElementWrapper
