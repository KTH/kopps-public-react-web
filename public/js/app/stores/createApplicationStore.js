/* eslint no-use-before-define: ["error", "nofunc"] */

// @ts-check

// eslint-disable-next-line no-unused-vars
import { observable } from 'mobx'
import commonStore from './commonStore'
import curriculumStore from './curriculumStore'
import studyProgrammeStore from './studyProgrammeStore'
import searchCoursesStore from './searchCoursesStore'
import appendix1Store from './appendix1Store'
import appendix2Store from './appendix2Store'
import literatureStore from './literatureStore'

export default createApplicationStore

const store = {
  /**
   * @property {string} message
   */
  message: 'Hallo',
  /**
   * @method
   * @param {string} text
   */
  setMessage,
  /**
   * @property {string} lastAdmissionTerm
   */
  lastAdmissionTerm: '',
  /**
   * @method
   * @param {string} lastAdmissionTerm
   */
  setLastAdmissionTerm,
  /**
   * @method
   * @param {map<string, {}>} programmes
   */
  setProgrammes,

  /**
   * @property {map<string, {}>} programmes
   */
  programmes: [],

  /**
   * @method
   * @param {[]} currentSchoolsWithDepartments
   */
  setCurrentSchoolsWithDepartments,

  /**
   * @property {[]} currentSchoolsWithDepartments
   */
  currentSchoolsWithDepartments: [],

  /**
   * @method
   * @param {[]} deprecatedSchoolsWithDepartments
   */
  setDeprecatedSchoolsWithDepartments,

  /**
   * @property {[]} deprecatedSchoolsWithDepartments
   */
  deprecatedSchoolsWithDepartments: [],
  /**
   * @property {string} departmentName
   */
  departmentName: '',
  /**
   * @method
   * @param {string} departmentName
   */
  setDepartmentName,

  /**
   * @property {[]} departmentCourses
   */
  departmentCourses: [],
  /**
   * @method
   * @param {[]} departmentCourses
   */
  setDepartmentCourses,
  /**
   * @property {string} programmeName
   */
  programmeName: '',
  /**
   * @method
   * @param {string} programmeName
   */
  setProgrammeName,
  /**
   * @property {string} programmeCode
   */
  programmeCode: '',
  /**
   * @method
   * @param {string} programmeCode
   */
  setProgrammeCode,
  /**
   * @property {[]} programmeTerms
   */
  programmeTerms: [],
  /**
   * @method
   * @param {[]} programmeTerms
   */
  setProgrammeTerms,
  /**
   * @property {number} lengthInStudyYears
   */
  lengthInStudyYears: 0,
  /**
   * @method
   * @param {number} lengthInStudyYears
   */
  setLengthInStudyYears,
  /**
   * @property {string} thisHostBaseUrl
   */
  thisHostBaseUrl: null,
}

function createApplicationStore(storeId) {
  switch (storeId) {
    case 'curriculum':
      return {
        ...commonStore,
        ...curriculumStore,
      }
    case 'searchCourses':
      return {
        ...commonStore,
        ...searchCoursesStore,
      }
    case 'objectives':
    case 'extent':
    case 'eligibility':
    case 'implementation':
      return {
        ...commonStore,
        ...studyProgrammeStore,
      }
    case 'appendix1':
      return {
        ...commonStore,
        ...studyProgrammeStore,
        ...appendix1Store,
      }
    case 'appendix2':
      return {
        ...commonStore,
        ...studyProgrammeStore,
        ...appendix2Store,
      }
    case 'literatureList':
      return {
        ...commonStore,
        ...literatureStore,
      }
    default:
      return {
        ...commonStore,
        ...store,
      }
  }
}

function setMessage(text = 'Happy coding!! :)') {
  this.message = text
}

function setProgrammes(programmes) {
  this.programmes = [...programmes]
}

function setCurrentSchoolsWithDepartments(currentSchoolsWithDepartments) {
  this.currentSchoolsWithDepartments = currentSchoolsWithDepartments
}

function setDeprecatedSchoolsWithDepartments(deprecatedSchoolsWithDepartments) {
  this.deprecatedSchoolsWithDepartments = deprecatedSchoolsWithDepartments
}

function setDepartmentName(departmentName) {
  this.departmentName = departmentName
}

function setDepartmentCourses(departmentCourses) {
  this.departmentCourses = departmentCourses
}

function setLastAdmissionTerm(lastAdmissionTerm) {
  this.lastAdmissionTerm = lastAdmissionTerm
}

function setProgrammeName(programmeName) {
  this.programmeName = programmeName
}

function setProgrammeCode(programmeCode) {
  this.programmeCode = programmeCode
}

function setProgrammeTerms(programmeTerms) {
  this.programmeTerms = programmeTerms
}

function setLengthInStudyYears(lengthInStudyYears) {
  this.lengthInStudyYears = lengthInStudyYears
}
