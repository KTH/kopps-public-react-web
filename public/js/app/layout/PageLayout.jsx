import React, { useEffect } from 'react'
import ReactDOM from 'react-dom'
import { Col, Row } from 'reactstrap'
import { Breadcrumbs } from '@kth/kth-reactstrap/dist/components/utbildningsinfo'

import { useStore } from '../mobx'
import MainMenu from '../components/MainMenu'

function renderBreadcrumbsIntoKthHeader(language) {
  const breadcrumbContainer = document.getElementById('breadcrumbs-header')
  if (breadcrumbContainer) {
    ReactDOM.render(<Breadcrumbs include="directory" language={language} />, breadcrumbContainer)
  }
}

function MainContent({ children }) {
  return (
    <Col>
      <main id="mainContent">{children}</main>
    </Col>
  )
}

function PageLayout({ children }) {
  const { language } = useStore()
  useEffect(() => renderBreadcrumbsIntoKthHeader(language))
  return (
    // Container in publicLayout.handlebars – begin
    <Row>
      <MainMenu />
      <MainContent>{children}</MainContent>
    </Row>
    // Container – end
  )
}

export default PageLayout
