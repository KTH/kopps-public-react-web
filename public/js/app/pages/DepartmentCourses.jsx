import React, { Fragment } from 'react'
import { Col, Row } from 'reactstrap'
import { Link, PageHeading, SortableTable } from '@kth/kth-reactstrap/dist/components/studinfo'

import Article from '../components/Article'
import FooterContent from '../components/FooterContent'

import { useStore } from '../mobx'
import i18n from '../../../../i18n'
import translate from '../../../../domain/translate'
import { courseLink } from '../util/links'

function codeCell(code) {
  const { language } = useStore()
  return {
    content: <Link href={courseLink(code, language)}>{code}</Link>,
    sortKey: code,
  }
}

function titleCell(code, title) {
  const { language } = useStore()
  return {
    content: <Link href={courseLink(code, language)}>{title}</Link>,
    sortKey: title,
  }
}

function compareCoursesBy(key) {
  return function compare(a, b) {
    if (a[key] < b[key]) {
      return -1
    }
    if (a[key] > b[key]) {
      return 1
    }
    return 0
  }
}

function sortAndParseCourses(courses) {
  courses.sort(compareCoursesBy('code'))
  const parsedCourses = courses.map(({ code, title, credits, creditUnitAbbr, level }) => [
    codeCell(code),
    titleCell(code, title),
    `${credits} ${creditUnitAbbr}`,
    level,
  ])
  return parsedCourses
}

function DepartmentsList() {
  const { language, departmentName, departmentCourses } = useStore()
  const t = translate(i18n, language)

  const headers = [
    t('department_course_code'),
    t('department_course_name'),
    t('department_course_credits'),
    t('department_course_educational_level'),
  ]
  const courses = sortAndParseCourses(departmentCourses)

  return (
    <>
      <Row>
        <Col>
          <PageHeading subHeading={departmentName}>{t('department_courses')}</PageHeading>
        </Col>
      </Row>
      <Row>
        <Col>
          <Article>
            <SortableTable headers={headers} data={courses} />
          </Article>
        </Col>
      </Row>
      <Row>
        <Col>
          <FooterContent />
        </Col>
      </Row>
    </>
  )
}

export default DepartmentsList
