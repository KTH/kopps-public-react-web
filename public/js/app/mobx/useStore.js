/* eslint no-use-before-define: ["error", "nofunc"] */

import createApplicationStore from '../stores/createApplicationStore'
import { getMobxManager } from './manager'

export default useStore

/**
 * Hook which can be used by any React component
 * within the related MobxStoreProvider.
 *
 * @param {string} [storeId]
 *
 *      Current state of application store
 */
function useStore(storeId = 'default') {
  const manager = getMobxManager(storeId)

  const isTestEnvironmentWithoutProviderComponent = manager == null
  if (isTestEnvironmentWithoutProviderComponent) {
    const dummyStoreForTesting = createApplicationStore()
    return dummyStoreForTesting
  }

  return manager.getContextHook()
}
