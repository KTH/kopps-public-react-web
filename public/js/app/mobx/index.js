/* eslint no-use-before-define: ["error", "nofunc"] */

import MobxStoreProvider from './MobxStoreProvider'
import useStore from './useStore'
import { compressStoreIntoJavascriptCode, uncompressStoreInPlaceFromDocument } from './compress'
// import { compressStoreIntoJavascriptCode, uncompressStoreInPlaceFromDocument } from './compressWithPako'

export { MobxStoreProvider, useStore, compressStoreIntoJavascriptCode, uncompressStoreInPlaceFromDocument }
