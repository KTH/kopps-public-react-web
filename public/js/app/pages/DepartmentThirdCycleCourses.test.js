/* eslint-disable react/prop-types */

import React from 'react'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { axe, toHaveNoViolations } from 'jest-axe'
import { StaticRouter } from 'react-router-dom/server'
import { MobxStoreProvider } from '../mobx'

import DepartmentCourses from './DepartmentCourses'
import ElementWrapper from '../components/ElementWrapper'
import PageLayout from '../layout/PageLayout'

import createApplicationStore from '../stores/createApplicationStore'
import getThirdCycleBreadcrumbs from '../config/thirdCycleBreadcrumbs'
import getThirdCycleDepartmentMenuData from '../config/thirdCycleDepartmentMenuData'

import commonSettings from '../config/mocks/mockCommonSettings'

expect.extend(toHaveNoViolations)

const applicationStore = createApplicationStore()

const testDepartmentName = {
  sv: 'ABE/Betongbyggnad',
  en: 'ABE/Concrete Structures',
}
const testDepartmentCourses = {
  sv: [
    {
      code: 'FAF3115',
      title: 'Betong och andra cementbaserade material',
      info: '',
      credits: '7,5',
      creditUnitLabel: 'Högskolepoäng',
      creditUnitAbbr: 'hp',
      level: 'Forskarnivå',
      state: 'ESTABLISHED',
    },
    {
      code: 'F1C5031',
      title: 'Dimensioneringsmetoder för armerade betongkonstruktioner',
      info: '',
      credits: '7,5',
      creditUnitLabel: 'Högskolepoäng',
      creditUnitAbbr: 'hp',
      level: 'Forskarnivå',
      state: 'CANCELLED',
      last_examination_semester: '20222',
    },
  ],
  en: [
    {
      code: 'FAF3115',
      title: 'Concrete and Other Cement Based Materials',
      info: '',
      credits: '7.5',
      creditUnitLabel: 'Credits',
      creditUnitAbbr: 'hp',
      level: 'Third cycle',
      state: 'ESTABLISHED',
    },
    {
      code: 'F1C5031',
      title: 'Design Methods for Reinforced Concrete Structures',
      info: '',
      credits: '7.5',
      creditUnitLabel: 'Credits',
      creditUnitAbbr: 'hp',
      level: 'Third cycle',
      state: 'CANCELLED',
      last_examination_semester: '20222',
    },
  ],
}

const WrapperDepartmentCourses = ({ lang }) => {
  applicationStore.setLanguage(lang)
  applicationStore.setBrowserConfig(commonSettings)
  applicationStore.setDepartmentName(testDepartmentName[lang])
  const updatedApplicationStore = {
    ...applicationStore,
  }
  return (
    <StaticRouter>
      <ElementWrapper
        exact
        path="/utbildning/forskarutbildning/kurser/org/AFF"
        createBreadcrumbs={store => ({
          include: 'university',
          items: getThirdCycleBreadcrumbs(store),
        })}
        applicationStore={updatedApplicationStore}
        component={DepartmentCourses}
        layout={PageLayout}
        createMenuData={store => ({
          selectedId: 'courses',
          ...getThirdCycleDepartmentMenuData(store),
        })}
      />
    </StaticRouter>
  )
}

const DepartmentCoursesWithLayout = ({ lang }) => {
  applicationStore.setLanguage(lang)
  applicationStore.setBrowserConfig(commonSettings)
  applicationStore.setDepartmentName(testDepartmentName[lang])
  applicationStore.setDepartmentCourses(testDepartmentCourses[lang])
  const departmentMenuData = getThirdCycleDepartmentMenuData(applicationStore)
  const breadcrumbsItems = getThirdCycleBreadcrumbs(applicationStore)

  const updatedApplicationStore = {
    ...applicationStore,
  }
  return (
    <StaticRouter>
      <MobxStoreProvider initCallback={() => updatedApplicationStore}>
        <PageLayout
          breadcrumbs={{
            include: 'university',
            items: breadcrumbsItems,
          }}
          menuData={{
            selectedId: 'courses',
            ...departmentMenuData,
          }}
        >
          <DepartmentCourses />
        </PageLayout>
      </MobxStoreProvider>
    </StaticRouter>
  )
}

describe('Render component DepartmentCourses within RouterWrapper for third cycle studies', () => {
  test('simple render', done => {
    render(<WrapperDepartmentCourses lang="en" />)
    done()
  })
})

describe('Render component DepartmentCourses within Layout for third cycle studies', () => {
  test('verify that page is accessible', async () => {
    const { container } = render(<DepartmentCoursesWithLayout lang="en" />)
    expect(await axe(container)).toHaveNoViolations()
  })
  test('get page header in English', () => {
    render(<DepartmentCoursesWithLayout lang="en" />)
    const h1Header = screen.getByRole('heading', { level: 1 })
    expect(h1Header).toHaveTextContent('Courses')
  })
  test('get page header in Swedish', () => {
    render(<DepartmentCoursesWithLayout lang="sv" />)
    const h1Header = screen.getByRole('heading', { level: 1 })
    expect(h1Header).toHaveTextContent('Kurser')
  })
  test('match to snapshot in English', done => {
    const { asFragment } = render(<DepartmentCoursesWithLayout lang="en" />)
    expect(asFragment()).toMatchSnapshot()
    done()
  })
  test('match to snapshot in Swedish', done => {
    const { asFragment } = render(<DepartmentCoursesWithLayout lang="sv" />)
    expect(asFragment()).toMatchSnapshot()
    done()
  })
})

describe('Render component DepartmentCourses and check its menu, content and links for third cycle studies', () => {
  test('check all links on the page in English', () => {
    render(<DepartmentCoursesWithLayout lang="en" />)
    const links = screen.getAllByRole('link')
    expect(links.length).toBe(7)
    // Menu links
    expect(links[0]).toHaveTextContent('Courses')
    expect(links[0].href).toStrictEqual('http://localhost/utbildning/forskarutbildning/kurser?l=en')
    // Main content links
    expect(links[1]).toHaveTextContent('F1C5031')
    expect(links[1].href).toStrictEqual('http://localhost/student/kurser/kurs/F1C5031?l=en')
    expect(links[2]).toHaveTextContent('Design Methods for Reinforced Concrete Structures')
    expect(links[2].href).toStrictEqual('http://localhost/student/kurser/kurs/F1C5031?l=en')

    expect(links[3]).toHaveTextContent('FAF3115')
    expect(links[3].href).toStrictEqual('http://localhost/student/kurser/kurs/FAF3115?l=en')
    expect(links[4]).toHaveTextContent('Concrete and Other Cement Based Materials')
    expect(links[4].href).toStrictEqual('http://localhost/student/kurser/kurs/FAF3115?l=en')
    // Footer links
    expect(links[5]).toHaveTextContent('Central study counseling')
    expect(links[5].href).toStrictEqual('https://www.kth.se/en/studies/master/general-study-counselling-1.621634')
    expect(links[6]).toHaveTextContent('kopps@kth.se')
    expect(links[6].href).toStrictEqual('mailto:kopps@kth.se')
  })
  test('check all links on the page in Swedish', () => {
    render(<DepartmentCoursesWithLayout lang="sv" />)
    const links = screen.getAllByRole('link')
    expect(links.length).toBe(7)
    // Menu links
    expect(links[0]).toHaveTextContent('Kurser')
    expect(links[0].href).toStrictEqual('http://localhost/utbildning/forskarutbildning/kurser')
    // Main content links
    expect(links[1]).toHaveTextContent('F1C5031')
    expect(links[1].href).toStrictEqual('http://localhost/student/kurser/kurs/F1C5031')
    expect(links[2]).toHaveTextContent('Dimensioneringsmetoder för armerade betongkonstruktioner')
    expect(links[2].href).toStrictEqual('http://localhost/student/kurser/kurs/F1C5031')

    expect(links[3]).toHaveTextContent('FAF3115')
    expect(links[3].href).toStrictEqual('http://localhost/student/kurser/kurs/FAF3115')
    expect(links[4]).toHaveTextContent('Betong och andra cementbaserade material')
    expect(links[4].href).toStrictEqual('http://localhost/student/kurser/kurs/FAF3115')
    // Footer links
    expect(links[5]).toHaveTextContent('Central studievägledning')
    expect(links[5].href).toStrictEqual('https://www.kth.se/studievagledning')
    expect(links[6]).toHaveTextContent('kopps@kth.se')
    expect(links[6].href).toStrictEqual('mailto:kopps@kth.se')
  })
})
