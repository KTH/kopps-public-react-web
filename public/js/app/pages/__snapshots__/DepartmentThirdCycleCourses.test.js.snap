// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render component DepartmentCourses within Layout for third cycle studies match to snapshot in English 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <nav
      aria-label="Sub menu"
      class="col navbar navbar-expand-lg navbar-light"
      id="mainMenu"
    >
      <div
        class="collapse navbar-collapse"
        id="navbarNav"
      >
        <ul
          class="nav"
        >
          <li
            class="parentLink"
          >
            <a
              href="/kopps-public/utbildning/forskarutbildning/kurser?l=en"
            >
              Courses
            </a>
          </li>
        </ul>
        <ul
          class="nav nav-list expandable"
        >
          <li
            class="nav-item ancestor"
          >
            <div>
              ABE/Concrete Structures
            </div>
          </li>
          <li
            class="nav-item leaf selected"
          >
            <a
              class="nav-link"
            >
              Courses
            </a>
          </li>
        </ul>
      </div>
    </nav>
    <div
      class="col"
    >
      <main
        id="mainContent"
      >
        <div
          class="row"
        >
          <div
            class="col"
          >
            <header
              role="presentation"
            >
              <h1
                aria-labelledby="page-heading page-sub-heading"
                id="page-heading"
              >
                Courses
              </h1>
              <p
                aria-hidden="true"
                class="page-sub-heading"
                id="page-sub-heading"
              >
                ABE/Concrete Structures
              </p>
            </header>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col"
          >
            <article
              class="article "
            >
              <table
                class="table"
              >
                <thead>
                  <tr>
                    <th
                      style="cursor: pointer; user-select: none;"
                    >
                      <div
                        style="display: flex; flex-direction: row;"
                      >
                        <p>
                          Course code
                        </p>
                        <p
                          style="opacity: 0;"
                        >
                          ▲
                        </p>
                      </div>
                    </th>
                    <th
                      style="cursor: pointer; user-select: none;"
                    >
                      <div
                        style="display: flex; flex-direction: row;"
                      >
                        <p>
                          Course name
                        </p>
                        <p
                          style="opacity: 0;"
                        >
                          ▲
                        </p>
                      </div>
                    </th>
                    <th
                      style="cursor: pointer; user-select: none;"
                    >
                      <div
                        style="display: flex; flex-direction: row;"
                      >
                        <p>
                          Scope
                        </p>
                        <p
                          style="opacity: 0;"
                        >
                          ▲
                        </p>
                      </div>
                    </th>
                    <th
                      style="cursor: pointer; user-select: none;"
                    >
                      <div
                        style="display: flex; flex-direction: row;"
                      >
                        <p>
                          Educational level
                        </p>
                        <p
                          style="opacity: 0;"
                        >
                          ▲
                        </p>
                      </div>
                    </th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td>
                      <a
                        href="/student/kurser/kurs/FAF3115?l=en"
                      >
                        FAF3115
                      </a>
                    </td>
                    <td>
                      <a
                        href="/student/kurser/kurs/FAF3115?l=en"
                      >
                        Concrete and Other Cement Based Materials
                      </a>
                    </td>
                    <td>
                      7.5 hp
                    </td>
                    <td>
                      Third cycle
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <a
                        href="/student/kurser/kurs/F1C5031?l=en"
                      >
                        F1C5031
                      </a>
                    </td>
                    <td>
                      <a
                        href="/student/kurser/kurs/F1C5031?l=en"
                      >
                        Design Methods for Reinforced Concrete Structures
                      </a>
                    </td>
                    <td>
                      7.5 hp
                    </td>
                    <td>
                      Third cycle
                    </td>
                  </tr>
                </tbody>
              </table>
            </article>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col"
          >
            <footer
              class="border-top mt-4 pt-1"
              id="articleFooter"
            >
              <address>
                Contact for questions regarding education at KTH: 
                <a
                  href="https://www.kth.se/studycounselling"
                >
                  Central study counseling
                </a>
                <br />
                Contact regarding technical matters on this page: 
                <a
                  href="mailto:kopps@kth.se"
                >
                  kopps@kth.se
                </a>
              </address>
            </footer>
          </div>
        </div>
      </main>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Render component DepartmentCourses within Layout for third cycle studies match to snapshot in Swedish 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <nav
      aria-label="Undermeny"
      class="col navbar navbar-expand-lg navbar-light"
      id="mainMenu"
    >
      <div
        class="collapse navbar-collapse"
        id="navbarNav"
      >
        <ul
          class="nav"
        >
          <li
            class="parentLink"
          >
            <a
              href="/kopps-public/utbildning/forskarutbildning/kurser"
            >
              Kurser
            </a>
          </li>
        </ul>
        <ul
          class="nav nav-list expandable"
        >
          <li
            class="nav-item ancestor"
          >
            <div>
              ABE/Betongbyggnad
            </div>
          </li>
          <li
            class="nav-item leaf selected"
          >
            <a
              class="nav-link"
            >
              Kurser
            </a>
          </li>
        </ul>
      </div>
    </nav>
    <div
      class="col"
    >
      <main
        id="mainContent"
      >
        <div
          class="row"
        >
          <div
            class="col"
          >
            <header
              role="presentation"
            >
              <h1
                aria-labelledby="page-heading page-sub-heading"
                id="page-heading"
              >
                Kurser
              </h1>
              <p
                aria-hidden="true"
                class="page-sub-heading"
                id="page-sub-heading"
              >
                ABE/Betongbyggnad
              </p>
            </header>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col"
          >
            <article
              class="article "
            >
              <table
                class="table"
              >
                <thead>
                  <tr>
                    <th
                      style="cursor: pointer; user-select: none;"
                    >
                      <div
                        style="display: flex; flex-direction: row;"
                      >
                        <p>
                          Kurskod
                        </p>
                        <p
                          style="opacity: 0;"
                        >
                          ▲
                        </p>
                      </div>
                    </th>
                    <th
                      style="cursor: pointer; user-select: none;"
                    >
                      <div
                        style="display: flex; flex-direction: row;"
                      >
                        <p>
                          Kursnamn
                        </p>
                        <p
                          style="opacity: 0;"
                        >
                          ▲
                        </p>
                      </div>
                    </th>
                    <th
                      style="cursor: pointer; user-select: none;"
                    >
                      <div
                        style="display: flex; flex-direction: row;"
                      >
                        <p>
                          Omfattning
                        </p>
                        <p
                          style="opacity: 0;"
                        >
                          ▲
                        </p>
                      </div>
                    </th>
                    <th
                      style="cursor: pointer; user-select: none;"
                    >
                      <div
                        style="display: flex; flex-direction: row;"
                      >
                        <p>
                          Utbildningsnivå
                        </p>
                        <p
                          style="opacity: 0;"
                        >
                          ▲
                        </p>
                      </div>
                    </th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td>
                      <a
                        href="/student/kurser/kurs/FAF3115"
                      >
                        FAF3115
                      </a>
                    </td>
                    <td>
                      <a
                        href="/student/kurser/kurs/FAF3115"
                      >
                        Betong och andra cementbaserade material
                      </a>
                    </td>
                    <td>
                      7,5 hp
                    </td>
                    <td>
                      Forskarnivå
                    </td>
                  </tr>
                  <tr>
                    <td>
                      <a
                        href="/student/kurser/kurs/F1C5031"
                      >
                        F1C5031
                      </a>
                    </td>
                    <td>
                      <a
                        href="/student/kurser/kurs/F1C5031"
                      >
                        Dimensioneringsmetoder för armerade betongkonstruktioner
                      </a>
                    </td>
                    <td>
                      7,5 hp
                    </td>
                    <td>
                      Forskarnivå
                    </td>
                  </tr>
                </tbody>
              </table>
            </article>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col"
          >
            <footer
              class="border-top mt-4 pt-1"
              id="articleFooter"
            >
              <address>
                Kontakt för frågor om utbildning på KTH: 
                <a
                  href="https://www.kth.se/studycounselling"
                >
                  Central studievägledning
                </a>
                <br />
                Teknisk kontakt om denna sida: 
                <a
                  href="mailto:kopps@kth.se"
                >
                  kopps@kth.se
                </a>
              </address>
            </footer>
          </div>
        </div>
      </main>
    </div>
  </div>
</DocumentFragment>
`;
